### ISSUE TITLE
can't run without pygmo

### ISSUE THREAD

**Issue by pentatonicfunk** [alan-turing-institute/AIrsenal#559]
```
Traceback (most recent call last):
  File "/Users/hendrawan/Green/AIrsenal/venv39/bin/airsenal_run_pipeline", line 5, in <module>
    from airsenal.scripts.airsenal_run_pipeline import run_pipeline
  File "/Users/hendrawan/Green/AIrsenal/airsenal/scripts/airsenal_run_pipeline.py", line 27, in <module>
    from airsenal.scripts.fill_transfersuggestion_table import run_optimization
  File "/Users/hendrawan/Green/AIrsenal/airsenal/scripts/fill_transfersuggestion_table.py", line 63, in <module>
    from airsenal.scripts.squad_builder import fill_initial_squad
  File "/Users/hendrawan/Green/AIrsenal/airsenal/scripts/squad_builder.py", line 7, in <module>
    from pygmo.core import sga
ModuleNotFoundError: No module named 'pygmo'
```
python 3.9. Mac M1

**Comment by pentatonicfunk** [comment]
using 3.8 also not working with different error

```
Traceback (most recent call last):
  File "/Users/hendrawan/Green/AIrsenal/venv3.8/bin/airsenal_run_pipeline", line 5, in <module>
    from airsenal.scripts.airsenal_run_pipeline import run_pipeline
  File "/Users/hendrawan/Green/AIrsenal/venv3.8/lib/python3.8/site-packages/airsenal/scripts/airsenal_run_pipeline.py", line 22, in <module>
    from airsenal.scripts.fill_db_init import check_clean_db, make_init_db
  File "/Users/hendrawan/Green/AIrsenal/venv3.8/lib/python3.8/site-packages/airsenal/scripts/fill_db_init.py", line 15, in <module>
    from airsenal.scripts.fill_player_table import make_player_table
  File "/Users/hendrawan/Green/AIrsenal/venv3.8/lib/python3.8/site-packages/airsenal/scripts/fill_player_table.py", line 16, in <module>
    from airsenal.scripts.fill_player_mappings_table import (
  File "/Users/hendrawan/Green/AIrsenal/venv3.8/lib/python3.8/site-packages/airsenal/scripts/fill_player_mappings_table.py", line 12, in <module>
    def load_mappings_data() -> list[list[str]]:
TypeError: 'type' object is not subscriptable
```

**Comment by rchan26** [comment]
hi @pentatonicfunk - I've just made a quick change to fix your second error with 3.8, can you see if this works now with 3.8?

as for your first error, did you follow the installation instructions on our [README](https://github.com/alan-turing-institute/AIrsenal/blob/main/README.md)? it should've installed `pygmo` and it should've worked, or are you having problems installing `pygmo` on your machine?

**Comment by rchan26** [comment]
I did a bit of digging on the `pygmo` error, and from #371, it seems that there is no pygmo support for 3.9 (see https://github.com/esa/pygmo2/issues/76, and #495) I think using the installation instructions on our README will install pygmo with Conda as a work around, and it should work, but you won't be able to pip install it - and so I see why that didn't work for you above, as it looks like you're using `venv`?

hopefully the fix above for the 3.8 error means that this will work for you on 3.8 though, so let me know if it doesn't!

but if you want to use 3.9, you'll have to use `conda` or `docker` I think

**Comment by rchan26** [comment]
realised last night from [this comment](https://github.com/alan-turing-institute/AIrsenal/issues/495#issuecomment-1212111173) that it shouldn't be possible to get this error as it should only load in `pygmo` if it's available. I think the error was re-introduced when we added type-hints, but the above PR #563 should fix this...

so after that is merged into main, you should be able to use with 3.9 too 😄

**Comment by pentatonicfunk** [comment]
yeap, after #562 went to main, it does now work with 3.8.

**Comment by griff-rees** [comment]
FYI: commit [baab616](https://github.com/alan-turing-institute/AIrsenal/commit/baab616b92a7c24092e9d6f929ae2fc08277709e) may indicate a way to run the model via `podman` or `docker` without `pygmo`

**Comment by rosseland** [comment]
running airsenal in docker fails on both main and develop due to missing pygmo on the optimize fase:

```
Prediction complete..
Unable to access FPL team API https://fantasy.premierleague.com/api/entry/xxx/event/1/picks/
Generating a squad..
Traceback (most recent call last):
  File "/airsenal/airsenal/framework/optimization_pygmo.py", line 10, in <module>
    import pygmo as pg
ModuleNotFoundError: No module named 'pygmo'

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "/root/.cache/pypoetry/virtualenvs/airsenal-SlSrXHQI-py3.9/lib/python3.9/site-packages/click/core.py", line 1130, in __call__
    return self.main(*args, **kwargs)
  File "/root/.cache/pypoetry/virtualenvs/airsenal-SlSrXHQI-py3.9/lib/python3.9/site-packages/click/core.py", line 1055, in main
    rv = self.invoke(ctx)
  File "/root/.cache/pypoetry/virtualenvs/airsenal-SlSrXHQI-py3.9/lib/python3.9/site-packages/click/core.py", line 1404, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/root/.cache/pypoetry/virtualenvs/airsenal-SlSrXHQI-py3.9/lib/python3.9/site-packages/click/core.py", line 760, in invoke
    return __callback(*args, **kwargs)
  File "/airsenal/airsenal/scripts/airsenal_run_pipeline.py", line 150, in run_pipeline
    new_squad_ok = run_make_squad(gw_range, fpl_team_id, dbsession)
  File "/airsenal/airsenal/scripts/airsenal_run_pipeline.py", line 248, in run_make_squad
    from airsenal.framework.optimization_pygmo import make_new_squad_pygmo
  File "/airsenal/airsenal/framework/optimization_pygmo.py", line 12, in <module>
    raise ModuleNotFoundError(
ModuleNotFoundError: Optional dependency pygmo not installed. If using conda run 'conda install pygmo'. If not see https://esa.github.io/pygmo2/install.html
```

**Comment by jack89roberts** [comment]
@rosseland Could you try on the develop branch now? I've added a workaround for it (though note I'm occasionally seeing some unrelated errors as well where the points prediction is failing due to a database error).
